File: solutions/zig/rna-transcription/1/rna_transcription.zig
const mem = @import("std").mem;

pub fn toRna(allocator: mem.Allocator, dna: []const u8) mem.Allocator.Error![]const u8 {
    const rna = try allocator.alloc(u8, dna.len);
    errdefer allocator.free(rna);

    for (dna, 0..) |n, idx| {
        switch (n) {
            'G' => rna[idx] = 'C',
            'C' => rna[idx] = 'G',
            'T' => rna[idx] = 'A',
            'A' => rna[idx] = 'U',
            else => continue,
        }
    }

    return rna;
}

File: solutions/javascript/rna-transcription/1/rna-transcription.js
var DnaTranscriber = function() {

}

DnaTranscriber.prototype.toRna = function(strand) {
  var map = {
    'G': 'C',
    'C': 'G',
    'T': 'A',
    'A': 'U'
  }

  rna = ''
  for(var i = 0; i < strand.length; i++) {
    if(!(strand[i] in map)) {
      throw new Error('Invalid input')
    }
    rna += map[strand[i]]
  }
  return rna
}

module.exports = DnaTranscriber